lib/rouge/lexers/kotlin.rb

```diff
@@ -73,50 +71,50 @@ class Kotlin < RegexLexer
         rule %r'"(\\\\|\\"|[^"\n])*["\n]'m, Str
         rule %r"'\\.'|'[^\\]'", Str::Char
         rule %r"[0-9](\.[0-9]+)?([eE][+-][0-9]+)?[flFL]?|0[xX][0-9a-fA-F]+[Ll]?", Num
-        rule %r/(@#{upper}#{name_backtick})/, Name::Decorator
-        rule %r'(#{upper}#{name_backtick})(<)' do
+        rule %r'(@#{class_name})', Name::Decorator
+        rule %r'(#{class_name})(<)' do
           groups Name::Class, Punctuation
           push :generic_parameters
         end
-        rule %r'(#{upper}#{name_backtick})', Name::Class
-        rule %r'(#{name_backtick})(?=\s*[({])', Name::Function
-        rule id, Name
+        rule %r'(#{class_name})', Name::Class
+        rule %r'(#{name})(?=\s*[({])', Name::Function
+        rule %r'#{name}', Name
```

Comment:
I like the visual consistency of interpolating the regex but this creates objects. Can you replace it with `name`? Same with the other instances below.

Reply:
Ah yes of course, sorry slipped my mind that the properties were regexes themselves. Will do that now!

Reply:
Done 👍